subprojects/core/src/main/java/org/gradle/internal/service/scopes/BuildSessionScopeServices.java

```diff
@@ -137,6 +138,10 @@ BuildOperationTrace createBuildOperationTrace(StartParameter startParameter, Bui
         return new BuildOperationTrace(startParameter, listenerManager);
     }
 
+    BuildOperationRecorder createBuildOperationRecorder(BuildOperationListenerManager listenerManager) {
```

Comment:
When I looked at this again, I think this can't be build-session scoped because it means we'd always miss "early" events when running in continuous build.

I know we don't produce build scans for continuous builds, but I think this should still work as if we did.